gammapy/maps/hpxnd.py

```diff
@@ -452,6 +453,47 @@ def smooth(self, width, kernel="gauss"):
 
         return self._init_copy(data=smoothed_data)
 
+    def convolve_wcs(self, kernel, use_fft=True, **kwargs):
+        """
+        Convolve map with a WCS kernel.
+
+        It projects the map into a WCS geometry, convolves with a WCS kernel and
+        projects back into the initial Healpix geometry.
+
+        If the kernel is two dimensional, it is applied to all image planes likewise.
+        If the kernel is higher dimensional it must match the map in the number of
+        dimensions and the corresponding kernel is selected for every image plane.
+
+        Parameters
+        ----------
+        kernel : `~gammapy.irf.PSFKernel` or `numpy.ndarray`
+            Convolution kernel. Must have the same geometry as
+            `self.to_wcs()`.
+        use_fft : bool
+            Use `scipy.signal.fftconvolve` if True (default)
+            and `ndi.convolve` otherwise.
+        kwargs : dict
+            Keyword arguments passed to `scipy.signal.fftconvolve` or
+            `ndi.convolve`.
+
+        Returns
+        -------
+        map : `HpxNDMap`
+            Convolved map.
+        """
+        if self.geom.width < 10*u.deg:
```

Comment:
I think the number "10 deg" should be introduce as a class level variable like:
```python3

class HpxNDMap:
    convolve_wcs_max_width
```
This allows users in general to overwrite it, in case they know what they are doing...

Reply:
Or maybe even better: alternatively we could just say all partial maps are supported and we make a statement in the docstring on the accuracy for larger maps >~10 deg

Reply:
I agree. My idea was that ideally we would have this method (ok for small maps) and then a full-sky map one that involves stacking a partial map into a full-sky map if needed (which I expect to be slower, but if it isn't then I need to rethink things), and if you try to use `convolve_wcs` on a partial map too large, you just get a warning suggesting that the other one is more suited or something. But it needs some more thought/work.